tests/src/python/test_qgsserver_wfs.py

```diff
@@ -285,6 +285,34 @@ def test_getfeature_post(self):
 """
         tests.append(('within4326FilterTemplate_post', within4326FilterTemplate.format("")))
 
+        # Check get feature within linear ring with srsName=EPSG:3857 (different from the project/layer)
+        # The coordinates are converted from the one in 4326
+        within3857FilterTemplate = """<?xml version="1.0" encoding="UTF-8"?>
+<wfs:GetFeature service="WFS" version="1.0.0" {} xmlns:wfs="http://www.opengis.net/wfs" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.opengis.net/wfs http://schemas.opengis.net/wfs/1.1.0/wfs.xsd">
+  <wfs:Query typeName="testlayer" srsName="EPSG:3857" xmlns:feature="http://www.qgis.org/gml">
+    <ogc:Filter xmlns:ogc="http://www.opengis.net/ogc">
+      <Within>
+        <PropertyName>geometry</PropertyName>
+        <Polygon xmlns="http://www.opengis.net/gml" srsName="EPSG:3857">
+          <exterior>
+            <LinearRing srsName="EPSG:3857">
```

Comment:
can you remove srsName here (and on line https://github.com/qgis/QGIS/pull/37035/commits/9370d94addf06f45d469733a7458222ead25309d#diff-61a1cac6f0799a1d603fac5b80d0f5ddR268 ) ? 
It's the same as the polygon anyway (and it's not supported).